src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -257,8 +257,49 @@ def to_internal_index(
             )
         )
 
+    def index_to_datetime(self, index: int) -> datetime | None:
+        """Convert the given index to the underlying datetime.
+
+        Index 0 corresponds to the oldest timestamp in the buffer. The
+        resulting datetime can be outside the range of the buffer.
+
+        Args:
+            index: Index to convert.
+
+        Returns:
+            Datetime index where the value for the given index can be found.
+                Or None if the buffer is empty.
+        """
+        if self.oldest_timestamp is None:
+            return None
+        assert self.newest_timestamp is not None
+        ref_ts = (
+            self.oldest_timestamp
+            if index >= 0
+            else self.newest_timestamp + self._sampling_period
+        )
+        return ref_ts + index * self._sampling_period
+
+    def _index_to_slice(
```

Comment:
Mild suggestion, since this is not really returning a slice but just indices (and the *covered* part seems to be key):

```suggestion
    def _to_covered_indices(
```